src/foam/dao/CachingDAO.js

```diff
@@ -68,7 +69,7 @@ foam.CLASS({
         var self = this;
         var cacheFilled = cache.removeAll().then(function() {
           // First clear cache, then load the src into the cache
-          return src.select(cache).then(function() {
+          return src.select(self.DAOSink.create({dao: cache})).then(function() {
```

Comment:
Should select check the type of the argument it got and automatically apply this if it's a DAO?